File: lecture3/node_demo/src/node_demo.cpp
#include <rclcpp/rclcpp.hpp>

#include "node_demo/node_demo_interface.hpp"

int main(int argc, char* argv[]) {
    rclcpp::init(argc, argv);

    auto node = std::make_shared<NodeDemoInterface>("node_demo_cpp");

    rclcpp::shutdown();
}

File: lecture3/node_demo/scripts/node_demo.py
#!/usr/bin/env python3

import rclpy
from node_demo.node_demo_interface import NodeDemoInterface


def main(args=None):
    rclpy.init(args=args)
    try:
        node = NodeDemoInterface("node_demo_py")
    except Exception as e:
        rclpy.logging.get_logger("node_demo_py").error(f"Error initializing node: {e}")
    finally:
        rclpy.shutdown()


if __name__ == "__main__":
    main()
